front/app/modules/commercial/content_builder/admin/containers/index.tsx

```diff
@@ -57,34 +130,97 @@ const ContentBuilderPage = ({ params: { projectId } }: WithRouterProps) => {
       iframeRef.current.contentWindow.postMessage(nodes, window.location.href);
   };
 
-  return (
-    <Box display="flex" flexDirection="column" w="100%">
-      <Editor isPreview={false} onNodesChange={handleEditorChange}>
-        <ContentBuilderTopBar
-          mobilePreviewEnabled={mobilePreviewEnabled}
-          setMobilePreviewEnabled={setMobilePreviewEnabled}
-        />
-        <Box
-          mt={`${stylingConsts.menuHeight}px`}
-          display={mobilePreviewEnabled ? 'none' : 'flex'}
-        >
-          <ContentBuilderToolbox />
-          <StyledRightColumn>
-            <Box width="1000px">
-              <ContentBuilderFrame editorData={editorData} />
-            </Box>
-          </StyledRightColumn>
-          <ContentBuilderSettings />
-        </Box>
-        <Box
-          justifyContent="center"
-          display={mobilePreviewEnabled ? 'flex' : 'none'}
-        >
-          <ContentBuilderMobileView projectId={projectId} ref={iframeRef} />
-        </Box>
-      </Editor>
+  const handleSelectedLocaleChange = ({
+    locale,
+    editorData,
+  }: {
+    locale: Locale;
+    editorData: SerializedNodes;
+  }) => {
+    if (selectedLocale && selectedLocale !== locale) {
+      setDraftData({ ...draftData, [selectedLocale]: editorData });
+    }
+
+    iframeRef.current &&
+      iframeRef.current.contentWindow &&
+      iframeRef.current.contentWindow.postMessage(
+        { selectedLocale: locale },
+        window.location.href
+      );
+
+    setSelectedLocale(locale);
+  };
+
+  const contentBuilderVisible =
+    featureEnabled && pathname.includes('admin/content-builder');
+
+  return contentBuilderVisible ? (
```

Comment:
We could combine this check with the `isNilOrError` check on line 110 and return null there so we don't have to check here.